src/core/tiledmesh/qgscesiumtilesdataprovider.cpp

```diff
@@ -20,45 +20,190 @@
+        if ( !rootRegion.isNull() )
+        {
+          mZRange = QgsDoubleRange( rootRegion.zMinimum(), rootRegion.zMaximum() );
+          // The latitude and longitude values are given in radians!
+
+          const double xMin = rootRegion.xMinimum() * 180 / M_PI;
+          const double xMax = rootRegion.xMaximum() * 180 / M_PI;
+          const double yMin = rootRegion.yMinimum() * 180 / M_PI;
+          const double yMax = rootRegion.yMaximum() * 180 / M_PI;
+          mExtent = QgsRectangle( xMin, yMin, xMax, yMax );
+        }
+      }
+      else
+      {
+        // TODO: handle box, sphere bounding volumes
+      }
+    }
+  }
+}
+
+
+//
+// QgsCesiumTilesDataProvider
+//
+
 QgsCesiumTilesDataProvider::QgsCesiumTilesDataProvider( const QString &uri, const ProviderOptions &providerOptions, ReadFlags flags )
   : QgsTiledMeshDataProvider( uri, providerOptions, flags )
+  , mShared( std::make_shared< QgsCesiumTilesDataProviderSharedData >() )
 {
+  QgsDataSourceUri dsUri;
+  dsUri.setEncodedUri( uri );
+  mAuthCfg = dsUri.authConfigId();
+  mHeaders = dsUri.httpHeaders();
+
+  mIsValid = init();
+}
 
+QgsCesiumTilesDataProvider::QgsCesiumTilesDataProvider( const QgsCesiumTilesDataProvider &other )
+  : QgsTiledMeshDataProvider( other )
+  , mIsValid( other.mIsValid )
+  , mAuthCfg( other.mAuthCfg )
+  , mHeaders( other.mHeaders )
+  , mShared( other.mShared )
+{
 }
 
 QgsCesiumTilesDataProvider::~QgsCesiumTilesDataProvider() = default;
 
+QgsCesiumTilesDataProvider *QgsCesiumTilesDataProvider::clone() const
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+  return new QgsCesiumTilesDataProvider( *this );
+}
+
+bool QgsCesiumTilesDataProvider::init()
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+
+  QgsDataSourceUri dsUri;
+  dsUri.setEncodedUri( dataSourceUri() );
+
+  const QString tileSetUri = dsUri.param( QStringLiteral( "url" ) );
+  if ( !tileSetUri.isEmpty() )
+  {
+    const QUrl url( tileSetUri );
+
+    QNetworkRequest request = QNetworkRequest( url );
+    QgsSetRequestInitiatorClass( request, QStringLiteral( "QgsCesiumTilesDataProvider" ) )
+    mHeaders.updateNetworkRequest( request );
+
+    QgsBlockingNetworkRequest networkRequest;
+    networkRequest.setAuthCfg( mAuthCfg );
+
+    switch ( networkRequest.get( request ) )
+    {
+      case QgsBlockingNetworkRequest::NoError:
+        break;
+
+      case QgsBlockingNetworkRequest::NetworkError:
+      case QgsBlockingNetworkRequest::TimeoutError:
+      case QgsBlockingNetworkRequest::ServerExceptionError:
+        // TODO -- error reporting
+        return false;
+    }
+
+    const QgsNetworkReplyContent content = networkRequest.reply();
+    const QByteArray raw = content.content();
+
+    // Parse data
+    QJsonParseError err;
+    const QJsonDocument doc = QJsonDocument::fromJson( raw, &err );
+    if ( doc.isNull() )
+    {
+      return false;
+    }
+    mShared->setTilesetContent( doc.object().toVariantMap() );
```

Comment:
I slightly lean towards the QVariantMap way because it allows functions which are reusable in Python. But I agree here performance may be an issue. I've switched to nlohmann but with QVariant wrappers in the utils class for use in PyQGIS.